### PR TITLE
ber_metaocaml: n107 -> n111

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes

The configure script has changed and no longer seems to provide a way
to specify the location of the x11 libraries or headers.  Therefore
these configure options have been dropped.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/ocaml/ber-metaocaml.nix [hunk 1]
```diff
@@ -34,9 +34,7 @@ stdenv.mkDerivation rec {
 
   prefixKey = "-prefix ";
   configureFlags = optionals useX11
-    [ "-x11lib" x11lib
-      "-x11include" x11inc
-      "-flambda"
+    [ "--enable-flambda"
```
Comment (vbgl) [inline pkgs/development/compilers/ocaml/ber-metaocaml.nix:37]: Why is this `flambda` option guarded by the `useX11` condition? It seems unrelated…

Comment (SuperSandro2000) [reply pkgs/development/compilers/ocaml/ber-metaocaml.nix:37]: The formatting is now also weird. This could easily be in one line.

Comment (a-m-joseph) [reply pkgs/development/compilers/ocaml/ber-metaocaml.nix]: > The formatting is now also weird. This could easily be in one line.

Fixed.

Comment (a-m-joseph) [reply pkgs/development/compilers/ocaml/ber-metaocaml.nix]: > Why is this `flambda` option guarded by the `useX11` condition? It seems unrelated…

I'm afraid I have no idea.  I was hesitant to make a functionality change in a version-bumping PR, but if you prefer I would be happy to remove the `--enable-flambda`.

**Review by ghost — COMMENTED** [review]
x
